compiler/rustc_middle/src/util/mod.rs

```diff
@@ -29,3 +34,13 @@ impl std::ops::Deref for Providers {
         &self.queries
     }
 }
+
+impl<'tcx> TyCtxt<'tcx> {
+    pub fn env_var(self, key: &'tcx OsStr) -> Result<&'tcx str, VarError> {
```

Comment:
Can use `AsRef<OsStr>` here, then the function will accept string literals.
You even use `key.as_ref()` below, which is unnecessary if the key is a non-generic `OsStr`.